mteb/tasks/Retrieval/code/MBPPRetrieval.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+
+class MBPPRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="MBPPRetrieval",
+        description="A code retrieval task based on 378 Python programming problems from MBPP (Mostly Basic Python Programming). Each query is a natural language description of a programming task (e.g., 'Write a function to find the shared elements from the given two lists'), and the corpus contains Python code implementations. The task is to retrieve the correct code snippet that solves the described problem. Queries are problem descriptions while the corpus contains Python function implementations with proper syntax and logic.",
+        reference="https://huggingface.co/datasets/embedding-benchmark/MBPP",
+        dataset={
+            "path": "embedding-benchmark/MBPP",
+            "revision": "12a7c6b3b7e985be9b9f81e2f06306ffd17cc5a4",
```

Comment:
Hmm this version does not seem to exist. I was able to find the latest one though:
```
586a1fd6a0c63fdeda3b49c0293559a81c79cdec
```